Updated contribution guidelines

clarkus opened the pull request:
## Changes
Updated guidelines on requesting reviews from product design and guidelines for contributing to product design.

## Checklist
- [x] Titles are in [sentence case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/sentence-case)
- [x] Feature names are in [title case](https://apastyle.apa.org/style-grammar-guidelines/capitalization/title-case)
- [x] Words are spelled using American English
- [x] I have checked out our [style guide](https://github.com/PostHog/posthog.com/blob/master/STYLEGUIDE.md)

paolodamico commented on contents/handbook/company/contributing-to-product-design.md:
```diff
@@ -0,0 +1,22 @@
+* Is your design consistent with the visual style we have established for the product? If you're deviating, be sure to provide some details on why and on what problems you're hoping to solve.
+* Have you considered pseudo states and other interaction patterns for your designs? Hover, focus, animation, etc.
+* Does your design follow basic UX best practices?
+* Does your design use clear language, correct grammar, and adhere to our [writing style](/handbook/company/communication#writing-style)?
```
When should these guidelines apply? I think there are cases where making quick mockups is the best approach. Example: If you're blocked by design and you're just putting something together for a new feature that will later go through design lifecycle.

clarkus replied:
These guidelines apply to any design that is being considered for implementation and use by the majority of users - note that experiments and isolated alpha features in production aren't really part of this. Sometimes we iterate quickly in code depending on the type of problem being solved. But if we are putting together a polished, complete, swiss watch-like experience, then this is the stuff to consider before hand off.

paolodamico commented on contents/handbook/company/working-with-product-design.md:
```diff
@@ -8,14 +8,17 @@ If your team doesn't have a dedicated designer, create an [issue](https://github
+1. Have you provided sufficient context for the problem you're trying to solve?
+2. Are there any technical constraints to be aware of?
+3. Is there any user feedback or other qualitative data that can inform the problem?
+4. How soon do you need a solution?
```
```suggestion
4. How soon do you need a solution? Be explicit if you are or will be blocked without the design.
```

paolodamico commented on contents/handbook/company/working-with-product-design.md:
```diff
@@ -8,14 +8,17 @@ If your team doesn't have a dedicated designer, create an [issue](https://github
 ### When to loop in design
 
-Because we hire self-starters, there is no expectation that every project should start by running through design _first_.
+Because we hire self-starters, there is no expectation that every project should start by running through design _first_. If you are contributing design, be sure to consult the [contribution guidelines](/handbook/company/contributing-to-product-design)
```
```suggestion
Because we hire self-starters, there is no expectation that every project should start by running through design _first_. If you are contributing to design, be sure to consult the [contribution guidelines](/handbook/company/contributing-to-product-design)
```

paolodamico reviewed (commented):
lgtm! made some suggestions inline

clarkus commented on contents/handbook/company/contributing-to-product-design.md:
```diff
@@ -0,0 +1,22 @@
+
+## Reviews &amp; critique
+
+In order to ensure users have a good experience, we review each others' work. When requesting feedback, be sure to consider the following guidelines:
```
Given the ratio of designers to non-designers, I've been optimizing for letting conversations happen where ever they happen. While they originate in various places, they always wind up with a figma design and comment thread, and a comment on a github issue or PR that shares the solution more broadly. 

This is absolutely an internal checklist for anyone contributing product design work - these are the things I am going to think about first when reviewing, so it makes sense to consider that before sharing. Do you think we need some kind of template or checklist? This is targeted towards contributors who don't already have a designer assigned to their team.

corywatilo commented:
I wrote a few notes of points that might be worth adding. Feel free to add where you see relevant and ignore the rest.

(Also some of what we have in these pages is old - like, pre-dedicated product designer, so we can definitely remove anything that doesn't still feel relevant!)

---

### Designers always offer free consultations

- This is especially true for teams without a dedicated Product Designer
- Syncing before starting a project is always strongly recommended. It helps us stay on the same page and makes sure we only step on each others toes (and not entire feet).
- Don’t feel like you *have* to craft a solution yourself. (What might feel complicated to you might be a breeze for us, so don’t feel like you need to try to solve it first. If we might be able to save you hours of agony, a quick Zoom might be well worth it.)

### Conversations happen wherever it’s relevant, but design work should always live in our Figma org

- Depending on a project’s stage, conversations might naturally occur in various places (Google Docs, issues, Figma comments, or pull requests). Given the ratio of designers to engineers, we’ll stick with this method for the foreseeable future.
- Regardless of where conversations happen, design decisions should make their way into finalized Figma mockups to serve as the single source of truth.

### ~Pull requests~ Figma mockups > Issues and comments

- Completed Figma mockups hold (almost) as much weight as pull requests. Documented designs in Figma should be considered the source of truth. Any feedback (including GitHub comments) should make its way into mockups to be considered final.
- If effort goes into making a mockup and gathering feedback, as much effort as possible should go into making the final product look like the mock.

paolodamico commented:
This workflow proposal (and Figma being the source of truth) will affect Engineering and as they own the features, I think they should weight in here.


While I understand the rationale, and the value of having the actual product be as close as possible to the mockups, I'm not sure how realistic will it be that in practice everything is perfectly modeled on Figma. Feature owners tend to make some decisions on the fly based on some constraints or new information. We'll want to move with higher quality for certain things and with speed in others.

corywatilo commented:
> I'm not sure how realistic will it be that in practice everything is perfectly modeled on Figma

Sure, understood. I think the point I was trying to make is: if a product designer creates a polished mock in Figma, an engineer should avoid diverging from it without consulting the designer (who would more than likely update the mock to reflect the changes).

clarkus commented:
I've updated the guidelines to specify that assets are considered "for implementation"... the idea is to not diverge from the solution provided by the designer unless there is a justifiable reason. Justifiable reasons could be unforeseen technical constraints, a better understanding of the core problem, updated business needs based on company goals, etc. Even then, the designer assigned to the team should be consulted to address the new context in the solution. This could happen in figma, in github issues, etc. The goal is to consult the assigned PD to ensure the solution meets the latest requirements and context, and that whatever solution is decided on is documented in github.
